hibernate-core/src/main/antlr/hql-sql.g

```diff
@@ -436,7 +436,7 @@ selectExpr
 	;
 
 count
-    : #(COUNT  { inCount = true; } ( DISTINCT { inCountDistinct = true; } | ALL )? ( aggregateExpr | ROW_STAR ) ) {
+    : #(COUNT  { inCount = true; } ( DISTINCT { inCountDistinct = true; } | ALL )? ( aggregateExpr | ROW_STAR | query) ) {
```

Comment:
FWIW this goes well beyond what JPA says should be supported.  Of course that does not mean we have to be constrained by not supporting things beyond what JPA defines...  Really none of the queries associated with this Jira should be supported according to JPA... Again, FWIW.

Here is what the spec's BNF says:

````
COUNT ( [ DISTINCT ] identification_variable | state_valued_path_expression | single_valued_object_path_expression )
...
state_valued_path_expression ::=
state_field_path_expression | general_identification_variable
````

I just worry about the use of subquery here.  I believe we have had problems with supporting subquery in various contexts of a query.  @gbadner does that ring any bells? @dreab8 @Naros What do y'all think?